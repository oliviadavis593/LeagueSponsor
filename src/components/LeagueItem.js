import React, { Component } from 'react';


class LeagueItem extends Component {
    render() {
        const { league_name, website, price, location } = this.props; 
        return(
            <div className='ls-league'>
                <h2 className='ls-league__name'>
                    <a href={website} target='blank'>{league_name}</a>
                </h2>
                <div className='ls-league__location'>
                    {location}
                </div>
                <div className='ls-league__price'>
                    {price}
                </div>
            </div>
        )
    }
}

export default LeagueItem; 