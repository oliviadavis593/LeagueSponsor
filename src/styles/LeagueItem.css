.ls-league {
    --item-height: 88px;
    background-color: #233142;
    padding: 8px 0px 8px 8px; 
    margin: 0 20px 15px -20px;
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.4);
    color: #fff; 
}

.ls-league__name a {
    color: #fff; 
    text-decoration: none; 
}

.ls-league__name:hover a {
    color: #fbc531;
    transition: 0.7s; 
}

.ls-league__location {
    margin-bottom: 10px; 
}

.ls-league__price {
    color: #fbc531;
}

@media only screen and (min-width: 720px) {
    .ls-league__name {
        font-size: 27px; 
    }

    .ls-league__location {
        font-size: 20px; 
    }

    .ls-league__price {
        font-size: 17px; 
    }
}