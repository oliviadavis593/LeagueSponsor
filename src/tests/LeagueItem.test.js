import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import LeagueItem from '../components/LeagueItem';

describe('LeagueItem Component', () => {
   it(`renders without crashing`, () => {
       const div = document.createElement('div')
       ReactDOM.render(<BrowserRouter><LeagueItem /></BrowserRouter>, div)
       ReactDOM.unmountComponentAtNode(div)
   })

   it('renders the UI as expected', () => {
       const tree = renderer
            .create(<BrowserRouter><LeagueItem /></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
   })
})