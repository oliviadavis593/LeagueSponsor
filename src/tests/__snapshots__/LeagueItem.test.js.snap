// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeagueItem Component renders the UI as expected 1`] = `
<div
  className="ls-league"
>
  <h2
    className="ls-league__name"
  >
    <a
      target="blank"
    />
  </h2>
  <div
    className="ls-league__location"
  >
    Location: 
  </div>
  <div
    className="ls-league__price"
  >
    Price: $
  </div>
</div>
`;
